build/configure.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+// Generate and execute makefile
+
+var utils = require('utils');
+
+var PRE_APP_BUILD_STEPS = [
+  'clean-stage-app',
+  'svoperapps',
+  'webapp-manifests',
+  'contacts-import-services',
+  'search-provider',
+  'keyboard-layouts',
+  'preferences',
+  'settings',
+  'webapp-shared',
+  'copy-common-files'
+];
+
+var POST_APP_BUILD_STEPS = [
+  'media-resolution',
+  'post-manifest',
+  'multilocale',
+  'copy-build-stage-data',
+  'webapp-optimize',
+  'webapp-zip'
+];
+
+var Makefile = function(filename) {
+  this.tasks = [];
+  this.filename = filename || this.FILENAME;
+  this.env = utils.getEnv('BUILD_CONFIG');
+};
+
+Makefile.prototype = {
```

Comment:
I would like you to comment about what's this class should do, and what every step is for. If we could have an example it's even better.